c7n/resources/rds.py

```diff
@@ -121,6 +121,19 @@ class resource_type(TypeInfo):
 
     filter_registry = filters
     action_registry = actions
+    policy_data = dict()
+
+    def __init__(self, ctx, data):
+        super(RDS, self).__init__(ctx, data)
+        if self.data:
```

Comment:
why are you setting a class attribute here?

Reply:
In general RDS policy runs, API (**describe_db_instance**) calls for twice at the runtime.

When the first API call (**describe_db_instance**) happens, **self.data** contains policy configuration data. So the describe API call gets pre-query filters and collects only the expected.

But on the second API call (**describe_db_instance**) the value on **self.data** is empty. So the **resources** method is not able to get the pre-query filter information from policy configuration data. Hence API call collects all the resources results and send the results to generic filters. so pre-query configuration given on policy is not consider and we are getting incorrect/inaccurate data.

Here only second API call gets into the considertion.

So to retain/hold the policy configuration data, I used class attribute.

Reply:
I have moved the if loop out of __init__ method to resources method